In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.tree import DecisionTreeClassifier
from sklearn import tree
from sklearn.metrics import classification_report
from sklearn.preprocessing import LabelEncoder

In [8]:
data = pd.read_csv("iris.csv")
print(data)

     sepal.length  sepal.width  petal.length  petal.width    variety
0             5.1          3.5           1.4          0.2     Setosa
1             4.9          3.0           1.4          0.2     Setosa
2             4.7          3.2           1.3          0.2     Setosa
3             4.6          3.1           1.5          0.2     Setosa
4             5.0          3.6           1.4          0.2     Setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  Virginica
146           6.3          2.5           5.0          1.9  Virginica
147           6.5          3.0           5.2          2.0  Virginica
148           6.2          3.4           5.4          2.3  Virginica
149           5.9          3.0           5.1          1.8  Virginica

[150 rows x 5 columns]


In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal.length  150 non-null    float64
 1   sepal.width   150 non-null    float64
 2   petal.length  150 non-null    float64
 3   petal.width   150 non-null    float64
 4   variety       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [10]:
print(type(data))
print(data.shape)
print(data.size)

<class 'pandas.core.frame.DataFrame'>
(150, 5)
750


In [12]:
data[data.duplicated(keep = False)]

sepal.length  sepal.width  petal.length  petal.width    variety
101           5.8          2.7           5.1          1.9  Virginica
142           5.8          2.7           5.1          1.9  Virginica

In [14]:
data.isnull().sum()


sepal.length    0
sepal.width     0
petal.length    0
petal.width     0
variety         0
dtype: int64

#### Observations
- There are 150 rows and 5 columns
- There are no null values
- There is one duplicated row(101 and 142 are duplicated)
- the x-columns are sepal.legth , sepal.width,petal.length and petal.width
- All the x-columns are continuous
- The y-column is"variety" (object) which is to be converted to categorical (to be pred)
- There are three flower categories (classes)